lb.tf

```diff
@@ -54,106 +178,65 @@ resource "exoscale_compute" "lb" {
-            "api_int_ip"     = local.privnet_api
-            "cluster_domain" = local.cluster_domain
-          }
-        ))
-      }
-    ],
-    "runcmd" = [
-      "while lsof -F p /var/lib/dpkg/lock 2>/dev/null; do echo \"Waiting for dpkg lock...\"; sleep 15; done",
-      "curl -Lo /tmp/floaty.deb https://github.com/vshn/floaty/releases/latest/download/floaty_linux_amd64.deb",
-      "dpkg -i /tmp/floaty.deb",
-      "systemctl restart keepalived",
-      "shutdown --reboot +1 'Reboot for system setup'"
-    ]
-  }))
+  user_data = format("#cloud-config\n%s", yamlencode(merge(
+    local.common_user_data,
+    {
+      "fqdn"             = local.instance_fqdns[count.index],
+      "hostname"         = random_id.lb[count.index].hex,
+      "manage_etc_hosts" = true,
+    },
+    // Override ec2_userdata fact with a clean copy of the userdata, as
+    // Exoscale presents userdata gzipped which confuses facter completely.
+    // TODO: check how we do this using server-up.
+    {
+      "write_files" = concat(local.common_write_files, [
+        {
+          path       = "/run/tmp/ec2_userdata_override.yaml"
+          "encoding" = "b64"
+          "content" = base64encode(yamlencode({
+            "ec2_userdata" = format("#cloud-config\n%s", yamlencode(merge(
+              local.common_user_data,
+              {
+                "fqdn"             = local.instance_fqdns[count.index],
+                "hostname"         = random_id.lb[count.index].hex,
+                "manage_etc_hosts" = true,
+                "write_files"      = local.common_write_files,
+              }
+            )))
+          }))
+        }
+      ])
+    }
+  )))
 
   lifecycle {
     ignore_changes = [
       template_id
     ]
   }
-}
 
-resource "exoscale_network" "lb_vrrp" {
-  # only create the lb-vrrp network when we're not deploying the cluster into
-  # a private network anyway.
-  count        = var.use_privnet ? 0 : 1
-  zone         = var.region
-  name         = "${var.cluster_id}_lb_vrrp"
-  display_text = "${var.cluster_id} LB VRRP network"
+  depends_on = [
+    null_resource.register_lb,
+    local_file.lb_hieradata[0]
+  ]
 }
 
 resource "exoscale_nic" "lb" {
   count      = var.lb_count
   compute_id = exoscale_compute.lb[count.index].id
   # Use cluster network if it's provisioned, lb_vrrp network otherwise
-  network_id = var.use_privnet ? exoscale_network.clusternet[0].id : exoscale_network.lb_vrrp[0].id
-  # Privnet CIDR IPs starting from .2 when using clusternet
-  ip_address = var.use_privnet ? cidrhost(var.privnet_cidr, 2 + count.index) : null
+  network_id = exoscale_network.clusternet.id
+  # Privnet CIDR IPs starting from .21
+  # (IPs .1,.2,.3 will be assigned by Puppet profile_openshift4_gateway)
+  ip_address = cidrhost(var.privnet_cidr, 21 + count.index)
+}
+
+resource "exoscale_domain_record" "lb" {
+  count  = var.lb_count
+  domain = exoscale_domain.cluster.id
+  name   = random_id.lb[count.index].hex
+  # TBD
```

Comment:
What is missing here?

Reply:
I initially wasn't sure if a TTL of 10 minutes is reasonable for the LB DNS records, but I think it's good enough for now.